nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3BinaryManager.java

```diff
@@ -478,21 +464,25 @@ public boolean fetchFile(String digest, File file) throws IOException {
                 log.debug("fetching blob " + digest + " from S3");
             }
             try {
-
-                ObjectMetadata metadata = amazonS3.getObject(
-                        new GetObjectRequest(bucketName, bucketNamePrefix + digest), file);
-                // check ETag
-                String etag = metadata.getETag();
-                if (!isEncrypted && !etag.equals(digest) && !ServiceUtils.isMultipartUploadETag(etag)) {
-                    log.error("Invalid ETag in S3, ETag=" + etag + " digest=" + digest);
-                    return false;
+                Download download = transferManager.download(new GetObjectRequest(bucketName, bucketNamePrefix + digest), file);
+                download.waitForCompletion();
+                // Check ETag it is by default MD5 if not multipart
+                if (!isEncrypted && !digest.equals(download.getObjectMetadata().getETag())) {
+                    // In case of multipart it will happen, verify the downloaded file
+                    String currentDigest = DigestUtils.md5Hex(new FileInputStream(file));
+                    if (!currentDigest.equals(digest)) {
+                        log.error("Invalid ETag in S3, currentDigest=" + currentDigest + " expectedDigest=" + digest);
+                        throw new IOException("Invalid S3 object, it is corrupted expected digest is " + digest + " got " + currentDigest);
+                    }
                 }
                 return true;
             } catch (AmazonClientException e) {
                 if (!isMissingKey(e)) {
                     throw new IOException(e);
                 }
                 return false;
+            } catch (InterruptedException e) {
```

Comment:
Don't catch `InterruptedException`, but if you really must, just do:
            
            Thread.currentThread().interrupt(); // reset interrupted status
            throw new RuntimeException(e);